plugins/CustomTriggerCandidateMaker.cpp

```diff
@@ -56,13 +55,49 @@ CustomTriggerCandidateMaker::CustomTriggerCandidateMaker(const std::string& name
 }
 
 void
-CustomTriggerCandidateMaker::init(const nlohmann::json& obj)
+CustomTriggerCandidateMaker::init(std::shared_ptr<appfwk::ModuleConfiguration> mcfg)
 {
+
+  auto mtrg = mcfg->module<appdal::CustomTriggerCandidateMaker>(get_name());
+  try {
+    // Get the outputs
+    for(auto con: mtrg->get_outputs()){
+      if(con->get_data_type() == datatype_to_string<triggeralgs::TriggerCandidate>()){
+        m_trigger_candidate_sink = get_iom_sender<triggeralgs::TriggerCandidate>(con->UID());
+        break;
+      }
+    }
+  } catch (const ers::Issue& excpt) {
+    throw dunedaq::trigger::InvalidQueueFatalError(ERS_HERE, get_name(), "input/output", excpt);
+  }
+
   m_time_sync_source = get_iom_receiver<dfmessages::TimeSync>(".*");
-  m_trigger_candidate_sink =
-    get_iom_sender<triggeralgs::TriggerCandidate>(appfwk::connection_uid(obj, "trigger_candidate_sink"));
+
+  m_conf = mtrg->get_configuration();
+
+  // Parsing configuration object into map
+  for (int i = 0; i < static_cast<int>(m_conf->get_trigger_types().size()); i++) {
+    std::pair<int, long int> temp_pair{ m_conf->get_trigger_types()[i], m_conf->get_trigger_intervals()[i] };
+    m_tc_settings.push_back(temp_pair);
+  }
+
+  print_config();
+
+  // This parameter controls how many new timestamps are calculated when needed
+  // Currently precalculates events for the next 60 seconds
+  m_sorting_size_limit = 60 * m_conf->get_clock_frequency_hz();
```

Comment:
clock frequency should really come from Session's detector_configuration but that's OK for now

Reply:
Agreed thank you. I created a git issue so this won't get forgotten.

I imagine this would be done by adding this inside `TriggerApplication.cpp` in `appdal` to the `StandaloneTriggerCandidateMaker` descriptive object (so user cannot change it themselves), and it would be copied from Session's detector_configuration to that `StandaloneTriggerCandidateMaker` from within `generate_modules` inside `TriggerApplication.cpp`?

Or is there a way of getting Session's detector_configuration from `std::shared_ptr<appfwk::ModuleConfiguration>` somehow?

Reply:
Not at the moment but I was wondering if a helper method in `ModuleConfiguration` would be a good idea in a different PR (https://github.com/DUNE-DAQ/readoutlibs/pull/152#discussion_r1507532071). Maybe this suggests it is a good idea.